dm/worker/worker.go

```diff
@@ -662,3 +651,194 @@ func (w *Worker) MigrateRelay(ctx context.Context, binlogName string, binlogPos
 	}
 	return nil
 }
+
+// copyConfigFromWorker copies config items from dm-worker to sub task
+func (w *Worker) copyConfigFromWorker(cfg *config.SubTaskConfig) {
+	cfg.From = w.cfg.From
+
+	cfg.Flavor = w.cfg.Flavor
+	cfg.ServerID = w.cfg.ServerID
+	cfg.RelayDir = w.cfg.RelayDir
+	cfg.EnableGTID = w.cfg.EnableGTID
+
+	// we can remove this from SubTaskConfig later, because syncer will always read from relay
+	cfg.AutoFixGTID = w.cfg.AutoFixGTID
+
+	// log config items, mydumper unit use it
+	cfg.LogLevel = w.cfg.LogLevel
+	cfg.LogFile = w.cfg.LogFile
+}
+
+// findSubTask finds sub task by name
+func (w *Worker) findSubTask(name string) *SubTask {
+	w.RLock()
+	defer w.RUnlock()
+
+	return w.subTasks[name]
+}
+
+func (w *Worker) restoreSubTask() error {
+	tasks := w.meta.LoadTaskMeta()
+
+	w.Lock()
+	defer w.Unlock()
+	w.subTasks = make(map[string]*SubTask)
+	for name, task := range tasks {
+		taskCfg := new(config.SubTaskConfig)
+		if err := taskCfg.Decode(string(task.Task)); err != nil {
+			return errors.Annotatef(err, "decode subtask config %s error in restoreSubTask", task.Task)
+		}
+
+		cfgDecrypted, err := taskCfg.DecryptPassword()
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		log.Infof("[worker] start sub task with config: %v", cfgDecrypted)
+
+		var st *SubTask
+		if task.GetStage() == pb.Stage_Running || task.GetStage() == pb.Stage_New {
+			st = NewSubTaskWithStage(cfgDecrypted, pb.Stage_New)
+			st.Run()
+		} else {
+			st = NewSubTaskWithStage(cfgDecrypted, task.Stage)
+		}
+
+		w.subTasks[name] = st
+	}
+
+	return nil
+}
+
+var maxRetryCount = 10
+
+func (w *Worker) handleTask() {
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+
+	retryCnt := 0
+
+Loop:
+	for {
+		select {
+		case <-w.ctx.Done():
+			return
+		case <-ticker.C:
+			w.Lock()
+			if w.closed.Get() == closedTrue {
+				w.Unlock()
+				return
+			}
+
+			opLog := w.meta.PeekLog()
+			if opLog == nil {
+				w.Unlock()
+				continue
+			}
+
+			log.Infof("start to execute operation ID, %d detail %+v", opLog.Id, opLog)
+
+			st, exist := w.subTasks[opLog.Task.Name]
+			var err error
+			switch opLog.Task.Op {
+			case pb.TaskOp_Start:
+				if exist {
+					err = errors.AlreadyExistsf("sub task %s", opLog.Task.Name)
+					break
+				}
+
+				if w.relayPurger.Purging() {
+					if retryCnt < maxRetryCount {
+						retryCnt++
+						log.Warnf("relay log purger is purging, cannot start sub task %s, would try again later", opLog.Task.Name)
+						w.Unlock()
+						continue Loop
```

Comment:
Is there any difference to use `continue Loop` and `continue` here?

Reply:
no,  it's a coding habits to make structure more clear